utils.go

```diff
@@ -0,0 +1,155 @@
+package legs
```

Comment:
there are a few different unrelated things in this file. consider splitting it up:
* simple setup
* the voucher stuff